components/devicecloud-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/GroupManagerService.java

```diff
@@ -0,0 +1,252 @@
+package org.wso2.carbon.device.mgt.iot.common.api;
+
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.group.common.Group;
+import org.wso2.carbon.device.mgt.group.common.GroupManagementException;
+import org.wso2.carbon.device.mgt.iot.common.GroupManagement;
+import org.wso2.carbon.user.core.Permission;
+
+import javax.ws.rs.*;
+import java.util.Date;
+import java.util.List;
+
+public class GroupManagerService {
+
+    @Path("/group")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean addGroup(@FormParam("name") String name, @FormParam("username") String username, @FormParam("description") String description) {
```

Comment:
SInce this is a post, I think it's better to associate all input values with one bean representing "group" rather than separate form params.